helios-client/src/main/java/com/spotify/helios/common/descriptors/RolloutOptions.java

```diff
@@ -31,20 +35,25 @@
 
   public static final long DEFAULT_TIMEOUT = TimeUnit.MINUTES.toSeconds(5);
   public static final int DEFAULT_PARALLELISM = 1;
+  public static final int DEFAULT_FAILURE_THRESHOLD = 1;
 
   private final long timeout;
   private final int parallelism;
   private final boolean migrate;
   private final boolean overlap;
-
-  public RolloutOptions(@JsonProperty("timeout") final long timeout,
-                        @JsonProperty("parallelism") final int parallelism,
-                        @JsonProperty("migrate") final boolean migrate,
-                        @JsonProperty("overlap") boolean overlap) {
+  private final int failureThreshold;
+
+  public RolloutOptions(
+      @JsonProperty("timeout") final long timeout,
+      @JsonProperty("parallelism") final int parallelism,
+      @JsonProperty("migrate") final boolean migrate,
+      @JsonProperty("overlap") boolean overlap,
+      @JsonProperty("failureThreshold") @Nullable final Integer failureThreshold) {
```

Comment:
Why `@Nullable Integer` and not just an int? Using `@DefaultValue` to set a default seems nicer, alternatively you can let it be 0 for existing deployment groups. This should be safe since we do a greater-than comparison (as opposed to a >=). x > 0 is only true if x is non-zero, which means one host failed -- which is the current behaviour.